pkg/quota/v1/install/update_filter.go

```diff
@@ -44,3 +51,17 @@ func DefaultUpdateFilter() func(resource schema.GroupVersionResource, oldObj, ne
 		return false
 	}
 }
+
+// hasResourcesChanged function to compare resources in container statuses
+func hasResourcesChanged(oldPod *v1.Pod, newPod *v1.Pod) bool {
+	if len(oldPod.Status.ContainerStatuses) != len(newPod.Status.ContainerStatuses) {
+		return false
+	}
+	for i, oldStatus := range oldPod.Status.ContainerStatuses {
+		newStatus := newPod.Status.ContainerStatuses[i]
```

Comment:
You can't assume that the old ContainerStatuses were in the same order. Use `podutil.GetContainerStatus(newPod.Status.ContainerStatuses, oldStatus.Name)`

Reply:
fixed